File: Lab1/app/src/main/java/com/example/lab1/MainActivity.java
package com.example.lab1;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        setupButtonListeners();
    }

    private void setupButtonListeners() {
        Button btnLayout1 = findViewById(R.id.btn_layout1);
        btnLayout1.setOnClickListener(v -> startActivity(new Intent(this, Layout1Activity.class)));

        Button btnLayout2 = findViewById(R.id.btn_layout2);
        btnLayout2.setOnClickListener(v -> startActivity(new Intent(this, Layout2Activity.class)));

        Button btnLayout3 = findViewById(R.id.btn_layout3);
        btnLayout3.setOnClickListener(v -> startActivity(new Intent(this, Layout3Activity.class)));

        Button btnLayout4 = findViewById(R.id.btn_layout4);
        btnLayout4.setOnClickListener(v -> startActivity(new Intent(this, Layout4Activity.class)));

        Button btnLayout5 = findViewById(R.id.btn_layout5);
        btnLayout5.setOnClickListener(v -> startActivity(new Intent(this, Layout5Activity.class)));
    }
}

File: Lab01/app/src/main/java/com/example/lab01/MainActivity.kt
package com.example.lab01

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        setupButtonListeners()
    }

    private fun setupButtonListeners() {
        findViewById<Button>(R.id.btn_layout1).setOnClickListener {
            startActivity(Intent(this, Layout1Activity::class.java))
        }

        findViewById<Button>(R.id.btn_layout2).setOnClickListener {
            startActivity(Intent(this, Layout2Activity::class.java))
        }

        findViewById<Button>(R.id.btn_layout3).setOnClickListener {
            startActivity(Intent(this, Layout3Activity::class.java))
        }

        findViewById<Button>(R.id.btn_layout4).setOnClickListener {
            startActivity(Intent(this, Layout4Activity::class.java))
        }

        findViewById<Button>(R.id.btn_layout5).setOnClickListener {
            startActivity(Intent(this, Layout5Activity::class.java))
        }
        
        findViewById<Button>(R.id.btn_lab4).setOnClickListener {
            startActivity(Intent(this, Lab4MainActivity::class.java))
        }
        
        findViewById<Button>(R.id.btn_lab5).setOnClickListener {
            startActivity(Intent(this, Lab5MainActivity::class.java))
        }
    }
}